S 61032 chore api/digests

mirvin opened the pull request.

mirvin committed:
S-61032 Created api/digests/es6/digestAdded.js

mirvin committed:
S-61032 Created api/digests/es6/digestCreate.js

mirvin committed:
S-61032 Created api/digests/es6/digestCommitsGet.js

mirvin committed:
S-61032 Created api/digests/es6/digestFormatAsHal.js

mirvin committed:
S-61032 Created api/digests/es6/digestGet.js

mirvin committed:
S-61032 Created api/digests/es6/digestInboxesFormatAsHal.js

mirvin committed:
S-60132 Created /api/digests/es6/digestsController.js

mirvin committed:
S-60132 Changed /api/digest/es6/digestsFormatAsHal.js

mirvin committed:
S-60132 Changed /api/digests/digestInboxesGet.js,
/api/digests/digestsFormatAsHal.js and /api/digests/digestsGet.js

JogoShugh commented on src/app/api/digests/es6/digestAdded.js:
```diff
@@ -0,0 +1,40 @@
+	validate(data) {
+		return schemaValidator.validate('digest', data, SCHEMA);	
+	}	
+};
```
Code looks great, can we fix up the whitespace on this one though

JogoShugh commented on src/app/api/digests/es6/digestCommitsGet.js:
```diff
@@ -0,0 +1,14 @@
+
```
Code looks good, just needs some whitespace fixups too

JogoShugh commented on src/app/api/digests/es6/digestCreate.js:
```diff
@@ -0,0 +1,23 @@
+
```
Leading blank line? Nix it 🗡

JogoShugh commented on src/app/api/digests/es6/digestFormatAsHal.js:
```diff
@@ -0,0 +1,27 @@
+export default (href, instanceId, data) => {
```
Code  is great! Whitespace, not so much 💃

JogoShugh commented on src/app/api/digests/es6/digestGet.js:
```diff
@@ -0,0 +1,9 @@
+import digestFormatAsHal from './digestFormatAsHal';
+import validateUUID from '../validateUUID';
+
+export default (req, res, next) => {
```
We can kill the `next`

JogoShugh commented on src/app/api/digests/es6/digestInboxesFormatAsHal.js:
```diff
@@ -0,0 +1,56 @@
+	};
+
+	function createInboxHyperMediaResult(instanceId, inbox) {
+	  var result = {
```
Should use `let` here for this mutable variable. Since we actually reassign it later, we won't use `const` here.

JogoShugh commented on src/app/api/digests/es6/digestInboxesFormatAsHal.js:
```diff
@@ -0,0 +1,56 @@
+	  return result;
+	}
+
+	inboxIds.forEach((inboxId) => {
```
We can replace the .forEach here with the `for (const inboxId of inboxIds)` syntax like we did in the other file

JogoShugh commented on src/app/api/digests/es6/digestInboxesGet.js:
```diff
@@ -0,0 +1,30 @@
+        res.hal(hypermedia);
+    }   
+    catch (ex) {
+      if (ex instanceof CSError.ProjectionNotFound){
```
Code looks great! Just some whitespace issues in this `try / catch` block

JogoShugh commented on src/app/api/digests/es6/digestsFormatAsHal.js:
```diff
@@ -0,0 +1,34 @@
+    }
+  }
+
+  function createDigestHyperMediaResult(digest) {
```
Can replace this with `const create...Result = digest => { ` arrow function syntax

mirvin committed:
S-60132 /api/digest/
digestAdded.js
digestCommitsGet.js
digestCreate.js
digestGet.js
digestInboxesFormatAsHal.js
digestInboxesGet.js
digestsFormatAsHal.js

mirvin committed:
S-60132 Changed api/digests/es6/digestFormatAsHal.js
